palantirnet/butler
# remove ruby, browsersync, linting, pretty much everything

See full notes [here](https://docs.google.com/a/palantir.net/document/d/1k0WlZTJkta1A6D6xng-DvRjVChpXmSW-XqiO9BAEzn8/edit?usp=sharing)

## Current Functionality
building sculpin and compiling sass.

### Removes:
* js linting
* Browsersync
* Ruby 
* sitespeed testing

### Includes:
* LibSass
* autoprefixer
* npm-breakpoint
* zen-grids
* npm-shrinkwrap
* stylelint
* bem-selector linter
* deployment to gh-pages

## Testing Instructions:
* pull Butler into a project skeleton and place it in the project root
* `vagrant up`
* `vagrant ssh`
* `cd /var/sites/{project}/butler`
* `npm install -g gulp`
* `npm install`
* `gulp`
* in a browser navigate to `http://{project}.local:8000/`
* You should see the styleguide there
* make some changes in your text editor to sass or views
* refresh browser
* Observe your styleguide has been updated
* Write some "incorrect" sass (example: `body { background:darkseagreen;` }
* Observe there are warnings that appear in your terminal
